app/scripts/lib/marketing-email-client.js

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A client to talk to the basket marketing email server
+ */
+
+'use strict';
+
+define([
+  'lib/xhr',
+  'lib/promise',
+  'lib/constants'
+], function (xhr, p, Constants) {
+
+  var NEWSLETTER_ID = Constants.MARKETING_EMAIL_NEWSLETTER_ID;
+
+  function MarketingEmailClient(options) {
+    options = options || {};
+
+    var self = this;
+
+    self._xhr = options.xhr || xhr;
+    self._baseUrl = options.baseUrl;
+  }
+
+  MarketingEmailClient.prototype = {
+    _request: function (path, type, accessToken, data, headers) {
+      var url = this._baseUrl + path;
+
+      return this._xhr.oauthAjax({
+        url: url,
+        type: type,
+        accessToken: accessToken,
+        data: data,
+        headers: headers
+      })
+      .fail(function () {
+        // TODO - no server is listening just yet. Remove this when ready.
+        // ignore the error until the server is ready.
+        return {};
+      });
+    },
+
+    getPreferencesUrl: function (accessToken) {
+      var self = this;
+      return self._request('/lookup-user', 'get', accessToken)
+        .then(function (response) {
+          if (! (response && response.token)) {
+            return null;
+          }
+
+          // TODO - find out the format of the real URL.
+          return self._baseUrl + '/user/' + response.token;
+        });
+    },
+
+    optIn: function (accessToken) {
+      return this._request('/subscribe', 'post', accessToken, {
+        //jshint camelcase: false
+        newsletter_id: NEWSLETTER_ID
```

Comment:
I did a bit of testing on these endpoints in production.  They seem to expect a trailing slash, like "/subscribe/" rather than "/subscribe".  The body here will need to be { "newsletters": NEWSLETTER_ID, "optin": "Y" }, with the assumption that the email will be pulled out of the authz header automatically.